alica_viewer/include/alica_viewer/alica_plan_parser.h

```diff
@@ -0,0 +1,277 @@
+#ifndef ALICA_VIEWER_ALICA_PLAN_PARSER_H
+#define ALICA_VIEWER_ALICA_PLAN_PARSER_H
+
+#include <SystemConfig.h>
+#include <engine/PlanRepository.h>
+#include <engine/SimplePlanTree.h>
+#include <engine/Types.h>
+#include <engine/containers/PlanTreeInfo.h>
+#include <engine/model/EntryPoint.h>
+#include <engine/model/Plan.h>
+#include <engine/model/State.h>
+#include <engine/model/Task.h>
+#include <engine/parser/PlanParser.h>
+#include <supplementary/AgentID.h>
+#include <supplementary/AgentIDManager.h>
+
+namespace alica
+{
+
+typedef std::unordered_map<const supplementary::AgentID*, std::shared_ptr<SimplePlanTree>, supplementary::AgentIDHash, supplementary::AgentIDEqualsComparator>
```

Comment:
Why the shared ptr?

Reply:
The children of SimplePlanTree are an unordered set of shared_pointers of SimplePlanTree. So I needed to make this a shared pointer so that it is compatible with the SimplePlanTree class.

Reply:
Yes, how do you feel about changing SimplePlanTree?

Reply:
I redid the PlanTree information for the viewer alone for now. We can take up the SimplePlanTree re-implementation later when we have more time.